src/lib/spinel/spinel_interface.hpp

```diff
@@ -58,6 +59,21 @@ class SpinelInterface
     typedef Hdlc::MultiFrameBuffer<kMaxFrameSize> RxFrameBuffer;
 
     typedef void (*ReceiveFrameCallback)(void *aContext);
+
+    /**
+     * This method indicates whether or not the frame is the Spinel SPINEL_CMD_RESET frame.
+     *
+     * @param[in] aFrame   A pointer to buffer containing the spinel frame.
+     * @param[in] aLength  The length (number of bytes) in the frame.
+     *
+     * @returns true if the frame is the Spinel SPINEL_CMD_RESET frame, false otherwise.
+     *
+     */
+    static bool IsSpinelResetCommand(const uint8_t *aFrame, uint16_t aLength)
+    {
+        static constexpr uint8_t kSpinelResetCommand[] = {SPINEL_HEADER_FLAG | SPINEL_HEADER_IID_0, SPINEL_CMD_RESET};
+        return (aLength == sizeof(kSpinelResetCommand)) && (memcmp(aFrame, kSpinelResetCommand, aLength) == 0);
```

Comment:
Relax the check just in case in future we add more stuff to the command (to stay backward compatible?)
```suggestion
        return (aLength >= sizeof(kSpinelResetCommand)) && (memcmp(aFrame, kSpinelResetCommand, aLength) == 0);
```